pkg/git/git.go

```diff
@@ -23,13 +27,29 @@ type Client struct {
 	dir      string
 }
 
+// Options holds options for cloning a git repository
+type Options struct {
+	URL    string
+	Branch string
+	User   string
+	Email  string
+}
+
 // NewGitClient returns a client that can perform git operations
 func NewGitClient(ctx context.Context, timeout time.Duration) *Client {
 	return &Client{
 		executor: executor.NewShellExecutor(ctx, timeout),
 	}
 }
 
+// NewGitClientDir returns a client that can perform git operations in an already cloned directory
+func NewGitClientDir(ctx context.Context, timeout time.Duration, path string) *Client {
```

Comment:
Unrelated to this PR (sorry to bring this up again), but contexts shouldn't be stored in objects which may have a long life.

Reply:
yeahp, we need to fix this at some point.